src/sys/output.cxx

```diff
@@ -153,4 +155,21 @@ ConditionalOutput output_error(Output::getInstance());
 ConditionalOutput output_verbose(Output::getInstance(), false);
 ConditionalOutput output(Output::getInstance());
 
+void backupfile(const char* filename) {
+  struct stat buffer;
+  if (stat(filename, &buffer) == 0){
+    char filename_[strlen(filename)+10];
```

Comment:
`char[]` -> `std::string`

And instead of `sprintf`, we can do `filename_ = filename + "~" + std::to_string(j);`